pyteal/ast/opup.py

```diff
@@ -58,7 +58,9 @@ class OpUp:
             )
     """
 
-    def __init__(self, mode: OpUpMode, target_app_id: Expr = None):
+    def __init__(
+        self, mode: OpUpMode, target_app_id: Expr = None, inner_fee: Expr = Int(0)
```

Comment:
I initially chose `inner_fee` as constructor without a real strong opinion but I agree that pattern you described is the most common.

Also, since an forgetful re-use of a previously constructed `OpUp` with certain `inner_fee` could expose the Smart Contract funds, I think that having it as method parameter forces more explicit usage, which I think is what we should optimize for looking at security.

So, I'll switch `inner_fee` from constructor to method parameter.

Reply:
@cusma https://github.com/algorand/pyteal/pull/553/commits/6b054a17957a22668ccc760bc3a4690b4f478d1f works for me.  

I'd like 1 more reviewer to chime in before considering the thread _resolved_.

Reply:
Speaking to @michaeldiamant and reading the code, I support this approach of requiring `inner_fee` in every invocation.